drivers/serial/uart_mspm0g3xxx.c

```diff
@@ -186,20 +180,13 @@ static void uart_mspm0g3xxx_isr(const struct device *dev)
 	const struct uart_mspm0g3xxx_config *config = dev->config;
 	struct uart_mspm0g3xxx_data *const dev_data = dev->data;
 
-	/* Get the pending interrupt */
-	int int_status = DL_UART_getEnabledInterruptStatus(
-		config->regs, UART_MSPM0_RX_INTERRUPTS | UART_MSPM0_TX_INTERRUPTS);
+	/* pending interrupt will be used by the provided callback */
+	dev_data->pending_interrupt = DL_UART_getPendingInterrupt(config->regs);
 
 	/* Perform callback if defined */
 	if (dev_data->cb) {
 		dev_data->cb(dev, dev_data->cb_data);
 	}
-
-	/*
-	 * Clear interrupts only after cb called, as Zephyr UART clients expect
-	 * to check interrupt status during the callback.
-	 */
-	DL_UART_clearInterruptStatus(config->regs, int_status);
```

Comment:
Who clears the interrupt now? Does `_getPendingInterrupt` read and clear?

Reply:
> Who clears the interrupt now? Does _getPendingInterrupt read and clear?

Yes, it will be cleared after reading it.

>  And someone has to clear bits within dev_data->pending_interrupt right? Or is this just based off the fact that it is only used in these callbacks and thus only accessed after a reassignment?

Yes, the other callbacks are only used from the ISR routine. It's stated in the documentation that calling those methods outside of the ISR is undefined behavior.

Reply:
It's alright then :)